service-base/src/main/java/org/eclipse/hono/service/metric/MicrometerBasedMetrics.java

```diff
@@ -431,4 +448,75 @@ private void trackDeviceConnectionDuration(final String tenantId, final long dev
                             return null;
                         }));
     }
+
+    @Override
+    public SendMessageSampler create(final String messageType) {
+        return new SendMessageSampler() {
+            @Override
+            public Sample start(final String tenantId) {
+
+                final Timer.Sample sample = Timer.start(registry);
+
+                return new Sample() {
+
+                    private void complete(final String ...tags) {
+                        sample.stop(registry.timer(METER_DOWNSTREAM_SENT, tags));
+                    }
+
+
+                    @Override
+                    public void accepted() {
+
+                        complete(
+                                "type", messageType,
+                                "success", "true"
+                        );
+
+                    }
+
+                    @Override
+                    public void rejected(final int errorCode) {
```

Comment:
if we use a single method *completed(String outcome)* instead of *accepted()* and *rejected(int)* then we can also omit the *success* tag ..